### ISSUE TITLE
BaseModeler Options is missing properties

### ISSUE THREAD

**Issue by BelindaFelpelViadee** [bpmn-io/bpmn-js#1882]
### Describe the Bug

The current BaseViewerOptions from BaseViewer.d.ts is the following:

```
export type BaseViewerOptions = {
  width?: number|string;
  height?: number|string;
  position?: string;
  container?: string|HTMLElement;
  moddleExtensions?: ModdleExtensions;
  additionalModules?: ModuleDeclaration[];
};
```

BaseModeler extends BaseViewer types. Currently, properties as
- linting
- propertiesPanel
- keyboard

are not type conform. This produces type errors when using bpmn-js in typescript.

### Steps to Reproduce

1. create a basic modeler in a typescript project
2. extend BpmnModeler constructor with e.g.
 
```
new BpmnModeler({
            container: canvasId,
            additionalModules: [
                BpmnPropertiesPanelModule,
                BpmnPropertiesProviderModule,
                lintModule,
               [...]
            ],
            linting: {
                bpmnlint: bpmnlintConfig,
                active: false,
            },
            propertiesPanel: {
                parent: $bpmnJsPanel,
            },
            keyboard: { bindTo: document },
        
        });
```
3. build project using webpack

### Expected Behavior

Typing definitions should be complete and handle extensions correctly.

### Environment

 - Browser: n/A
 - OS: Windows 10
 - Library version: 
    - "bpmn-js": "12.0.0"
    - "webpack": "5.77.0"
    - "bpmn-js-properties-panel": "1.20.3"
    - "bpmn-js-bpmnlint": "0.20.1"
    - "bpmnlint": "8.2.0",

**Comment by nikku** [comment]
Thanks for opening this issue. We clearly need to make our types more open for extension. 

CC @philippfromme

**Comment by nikku** [comment]
Fixed with `bpmn-js@12.1.0`.
